### PR TITLE
Harden settings service.

### PR DESCRIPTION
**Pull request by pilhuhn**
Especially when notification system can't be reached.

Signed-off-by: Heiko W. Rupp <hrupp@redhat.com>

cc @karelhala

### CODE REVIEW

**Comment by theute** [comment]
Unable to do a real review, but definitely review to give it a try once merged

File: src/main/java/com/redhat/cloud/custompolicies/app/rest/SettingsService.java [hunk 1]
```diff
@@ -57,43 +57,64 @@
...
+
     values.username = user.getName();
     values.accountId = user.getAccount();
     SettingsValues tmp = SettingsValues.findById(user.getName());
     if (tmp != null) {
       tmp.immediateEmail = values.immediateEmail;
       tmp.dailyEmail = values.dailyEmail;
     } else {
-      values.persistAndFlush();
+      values.persist();
```
Comment (burmanm) [inline src/main/java/com/redhat/cloud/custompolicies/app/rest/SettingsService.java]: This still persists changes even if the store fails?

Comment (pilhuhn) [reply src/main/java/com/redhat/cloud/custompolicies/app/rest/SettingsService.java]: Should not, as there is a `@Transactional` annotation

Comment (burmanm) [reply src/main/java/com/redhat/cloud/custompolicies/app/rest/SettingsService.java]: Line 97 has ``catch (Exception e) {`` thus the JTA doesn't get any exception. So, what would rollback?

File: src/main/java/com/redhat/cloud/custompolicies/app/rest/SettingsService.java [hunk 2]
```diff
@@ -57,43 +57,64 @@
...
+        notifications.removeNotification("custom-policies-daily-mail", user.getRawRhIdHeader());
+      }
+      builder = Response.ok();
     }
-    else {
-      notifications.removeNotification("custom-policies-daily-mail", user.getRawRhIdHeader());
+    catch (Exception e) {
+      builder = Response.serverError().entity("Storing of settings in notification service failed.");
+      builder.type("text/plain");
+      System.err.println("Storing settings failed: " + e.getMessage());
```
Comment (burmanm) [inline src/main/java/com/redhat/cloud/custompolicies/app/rest/SettingsService.java]: Should probably be a logger line?

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -27,7 +27,7 @@ engine.skip=false
 rbac/mp-rest/url=https://ci.cloud.redhat.com
 
 # Notifications server
-notifications/mp-rest/url=http://custom-policies-notifications:8080
+notifications/mp-rest/url=http://custom-policies-notifications:8000
```
Comment (theute) [inline src/main/resources/application.properties]: How about we expose port 8080 like for other containers ?
I guess we just need to change:
https://github.com/RedHatInsights/e2e-deploy/blob/master/templates/custom-policies/custom-policies-notifications.yml#L15

Comment (theute) [reply src/main/resources/application.properties]: I made a PR here: https://github.com/RedHatInsights/e2e-deploy/pull/1242
It that gets approved, I can already test I think

Comment (pilhuhn) [reply src/main/resources/application.properties]: I am not opposed to that. Having 8080 everywhere makes it harder locally when all services are deployed 'bare metal'. cc @burmanm

**Comment by karelhala** [comment]
Error codes and return values look good to me!

File: src/main/resources/application.properties (hunk 3 above)
Comment (burmanm) [reply src/main/resources/application.properties]: Well, container can redirect ports so it doesn't matter - that PR didn't change the local port.

Comment (theute) [reply src/main/resources/application.properties]: So it's running on port 8080 now, we can change back if you prefer.

Comment (pilhuhn) [reply src/main/resources/application.properties]: What is the conclusion here? Do I change the default in here or not?

Comment (theute) [reply src/main/resources/application.properties]: it's listening on 8080 now

File: src/main/java/com/redhat/cloud/custompolicies/app/rest/SettingsService.java (hunk 1 above)
Comment (pilhuhn) [reply src/main/java/com/redhat/cloud/custompolicies/app/rest/SettingsService.java]: You are right. I changed this now.

**Pull request merged**